spacetime/tests/test_linear_scalar.py

```diff
@@ -133,4 +134,85 @@ def _march():
             self.assertEqual(self.svr.get_so0(0).ndarray.tolist(),
                              svr2.get_so0(0).ndarray.tolist())
 
+
+class LinearScalarGridTestTC(unittest.TestCase):
+    """
+    Compare linear solver's solution wtih exact solution,
+    because solutions of computation field is a vector,
+    therefore compared the norm of difference bewteen solver's solution
+    and exact solution.
+
+    By comparing norm of difference bewteen solver's solution and exact 
+    solution we can check if the linear solver is work properly
+    or check if the solver's mathematical model is correct.
+    """
+    
+    @staticmethod
+    def _build_solver(resolution):
+        grid = libst.Grid(0, 4 * 2 * np.pi, resolution)
+        cfl = 0.99
+        dx = (grid.xmax - grid.xmin) / grid.ncelm
+        dt = dx * cfl
+        svr = libst.LinearScalarSolver(grid=grid, time_increment=dt)
+
+        # Initialize
+        for e in svr.selms(odd_plane=False):
+            if e.xctr < 2 * np.pi or e.xctr > 4 * np.pi:
+                v = 0
+                dv = 0
+            else:
+                v = np.sin(e.xctr)
+                dv = np.cos(e.xctr)
+            e.set_so0(0, v)
+            e.set_so1(0, dv)
+
+        svr.setup_march()
+
+        return svr
+
+    @staticmethod
+    def _exact_solution(svr, iter_num, gfun):
+        v = []
+        for e in svr.selms(odd_plane=False):
+            x = (
+                e.xctr - iter_num * svr.dt
+            ) % svr.grid.xmax  # dealt with boundary treatment
+            if x < 2 * np.pi or x > 4 * np.pi:
+                v.append(0)
+            else:
+                v.append(gfun(x))
+        return np.array(v)
+
+    def setUp(self):
```

Comment:
Is this function redundant?  If it is, please remove it.

Reply:
Yes It is.
I forgot to remove it since I rewrite this test case.
I remove it in https://github.com/yungyuc/turgon/pull/32/commits/e9f9ca760bcebaa4220977bc9639b22a29cc8b40